woodchipper/http/flask.py

```diff
@@ -24,6 +24,7 @@ def wrapped_full_dispatch_request(self):
                 "body_size": request.content_length,
                 "method": request.method,
                 "path": request.base_url,
+                "url_rule": request.url_rule.rule,
```

Comment:
Darn, I thought I had covered that. Yes, I think so. A 404 scenario where no rule matches means this could attribute error trying to access `.rule` I think? `request.url_rule.rule if hasattr(request, "url_rule") else None` should do it.

Even if Flask exits before the context is opened we may refactor that in the future and have the context open earlier. I'll get in a test for an unmatched URL rule for that scenario to prevent a future regression.